component/async/sqs/sqs.go

```diff
@@ -135,9 +136,9 @@ type Factory struct {
 	queueName         string
 	queue             sqsiface.SQSAPI
 	queueURL          string
-	maxMessages       int64
-	pollWaitSeconds   int64
-	visibilityTimeout int64
+	maxMessages       *int64
+	pollWaitSeconds   *int64
+	visibilityTimeout *int64
```

Comment:
I agree no real need for this change. AWS has decided for some crazy reason to use only pointer which is spread throughout the SDK. No need to follow that craziness.

Reply:
There's a good reason for this guys. By passing nil to the AWS SDK you tell it to use the defaults configured on the queue itself, that we set in Terragrunt.

Reply:
I can see the why, but using pointers makes everything a good candidate for heap allocation. The escape analysis might yield a heap allocation because of this way... So only use pointers if the semantics warrant for it (modification etc.) and we can handle it at the SDK edge correctly.

Reply:
How would you implement this? We need to be able to differentiate between 0 and nil for pollWaitSeconds, because it's two entirely different things. If we don't have a pointer in the builder and the component, we would need additional boolean fields, unless you have a different suggestion.

Reply:
Did it not work before?

Reply:
Unless you can specify nil in the Builder, there is no way you can configure the consumer to use the queue's own defaults rather than overriding them.